pyogrio/_err.pyx

```diff
@@ -1,25 +1,26 @@
-# ported from fiona::_err.pyx
-from enum import IntEnum
+"""Error handling code for GDAL/OGR.
+
+Ported from fiona::_err.pyx
+"""
+
+import contextlib
 import warnings
+from contextvars import ContextVar
+from itertools import zip_longest
 
 from pyogrio._ogr cimport (
     CE_None, CE_Debug, CE_Warning, CE_Failure, CE_Fatal, CPLErrorReset,
     CPLGetLastErrorType, CPLGetLastErrorNo, CPLGetLastErrorMsg, OGRErr,
-    CPLErr, CPLErrorHandler, CPLDefaultErrorHandler, CPLPushErrorHandler)
-
-
-# CPL Error types as an enum.
-class GDALError(IntEnum):
-    none = CE_None
-    debug = CE_Debug
-    warning = CE_Warning
-    failure = CE_Failure
-    fatal = CE_Fatal
+    CPLErr, CPLErrorHandler, CPLDefaultErrorHandler, CPLPopErrorHandler,
+    CPLPushErrorHandler, CPLPushErrorHandlerEx)
 
+_ERROR_STACK = ContextVar("error_stack")
```

Comment:
Do you know the reason for using a ContextVar here? (and not a thread-local variable) from quickly reading up on it, it seems this is mostly used to have something similar to thread-local storage but in async / coroutines?

Reply:
No, I didn't really look into it in that detail...